numpy/core/numeric.py

```diff
@@ -650,23 +653,26 @@ def require(a, dtype=None, requirements=None):
     if not requirements:
         return asanyarray(a, dtype=dtype)
 
+    requirements = set(requirements)
```

Comment:
Is this really what we want?  If we only look at first letters anything that has the right first letter will work, seems like a place to be a little stricter.